src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/PrisonerSearchApiClientTest.kt

```diff
@@ -0,0 +1,93 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app
+
+import mu.KotlinLogging
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.RepeatedTest
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.Prisoner
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.PrisonerSearchApiClient
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.testcontainers.LocalStackContainer
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.testcontainers.LocalStackContainer.setLocalStackProperties
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.testcontainers.PostgresContainer
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Test class to demonstrate possible problem with prisoner-search-api when calling it with paged requests
+ */
+@SpringBootTest
+@ActiveProfiles("demo-prisoner-search-api-paging-bug")
+class PrisonerSearchApiClientTest {
+
+  companion object {
+    private const val PRISON_ID = "WDI"
+
+    private val postgresContainer = PostgresContainer.instance
+
+    private val localStackContainer = LocalStackContainer.instance
+
+    @JvmStatic
+    @DynamicPropertySource
+    fun configureTestContainers(registry: DynamicPropertyRegistry) {
+      postgresContainer?.run {
+        registry.add("spring.datasource.url", postgresContainer::getJdbcUrl)
+        registry.add("spring.datasource.username", postgresContainer::getUsername)
+        registry.add("spring.datasource.password", postgresContainer::getPassword)
+        registry.add("spring.datasource.placeholders.database_update_password", postgresContainer::getPassword)
+        registry.add("spring.datasource.placeholders.database_read_only_password", postgresContainer::getPassword)
+        registry.add("spring.flyway.url", postgresContainer::getJdbcUrl)
+        registry.add("spring.flyway.user", postgresContainer::getUsername)
+        registry.add("spring.flyway.password", postgresContainer::getPassword)
+      }
+      localStackContainer?.also { setLocalStackProperties(it, registry) }
+    }
+  }
+
+  @Autowired
+  private lateinit var prisonerSearchApiClient: PrisonerSearchApiClient
+
+  @Disabled
+  @RepeatedTest(100)
+  fun `test paged operation`() {
+    // First call prisoner-search-api to get all prisoners in BXI as a single page
+    val allPrisonersInOneCall = getAllPrisonersInPrison(PRISON_ID, 9999).map { it.prisonerNumber }
```

Comment:
This is basically our test control sample - we call the prisoner-search-api with a large page size, so it has the effect of calling it once as a non-paged operation.
It returns a list (which could contain duplicates)
To ensure all elements in the list are unique we assert that the list size is equal to the size of the list as a set.

This is our happy path / control sample.